### ISSUE TITLE
Impossible to select non-DOMElements

### ISSUE THREAD

**Issue by EdgarPE** [symfony/symfony#16933]
This patch: https://github.com/symfony/symfony/commit/9f362a12f616984676182f69aab18c662b47445d introduced a serious limitation, namely one can not select non-DOMElements. For example, now it is impossible to select only the text content of a specific node, like this: `$node->filterXPath('//text()')`
The selected node never gets returned because of this:
```php
if ($subNode->nodeType === 1) {
    $crawler->add($subNode);
}
```

I suggest a solution, where adding non-DOMElements are prevented but selecting one is possible. Is the community open for that? If yes, I am willing to work on it.

**Comment by jakzal** [comment]
@EdgarPE selecting and adding non-DOMElements is the same in this context, as crawler's filtering methods return crawler instances. Nodes are added to a new crawler instance, and then returned. 

See #16058 for reasons why this has been done.

**Comment by EdgarPE** [comment]
@jakzal Thank you, for your insights. Based on your comment I just made a Pull Request: https://github.com/symfony/symfony/pull/17021

**Comment by armpogart** [comment]
So is it expected behavior that now I can do `$node->filterXPath('//div')` but not `$node->filterXPath('//div/text()')`. My application was using `/text()` xpath selector intensively, what are you suggesting - to use older version? Will full xpath syntax be supported in any future version?

**Comment by EdgarPE** [comment]
Hopefully the Pull request will be merged soon. If you need to update to 2.8 or 3.0 now, one workaround is to select the supported node (//div), get the DOMElement object, then than use xpath to filter the DOMText object. It's not ideal, but it works.

**Comment by armpogart** [comment]
Thanks for your quick response, I will be glad if you also show me how to get the `DOMElement` object and use xpath on it. I don't get it, isn't filtered node always of type `Symfony\Component\DomCrawler\Crawler`? I couldn't find any way there to get the DOMElement.

Also does that mean if your PR will be merged `/text()` will work on 2.8 and 3.* also or 3.* will break the compatibility?

**Comment by stof** [comment]
@armpogart ``->getNode(0)`` gives you the underlying DOM node (change the argument to get other indexes of course). And the iterator also gives you DOM nodes.
